internal/services/storage/storage_account_local_user_resource.go

```diff
@@ -0,0 +1,528 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2021-09-01/storage"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type LocalUserResource struct{}
+
+var _ sdk.ResourceWithUpdate = LocalUserResource{}
+
+type PermissionsModel struct {
+	Create bool `tfschema:"create"`
+	Delete bool `tfschema:"delete"`
+	List   bool `tfschema:"list"`
+	Read   bool `tfschema:"read"`
+	Write  bool `tfschema:"write"`
+}
+type PermissionScopeModel struct {
+	Permissions  []PermissionsModel `tfschema:"permissions"`
+	ResourceName string             `tfschema:"resource_name"`
+	Service      string             `tfschema:"service"`
+}
+type SshAuthorizedKeyModel struct {
+	Description string `tfschema:"description"`
+	Key         string `tfschema:"key"`
+}
+type LocalUserModel struct {
+	HomeDirectory      string                  `tfschema:"home_directory"`
+	Name               string                  `tfschema:"name"`
+	Password           string                  `tfschema:"password"`
+	PermissionScope    []PermissionScopeModel  `tfschema:"permission_scope"`
+	Sid                string                  `tfschema:"sid"`
+	SshAuthorizedKey   []SshAuthorizedKeyModel `tfschema:"ssh_authorized_key"`
+	SshKeyEnabled      bool                    `tfschema:"ssh_key_enabled"`
+	SshPasswordEnabled bool                    `tfschema:"ssh_password_enabled"`
+	StorageAccountName string                  `tfschema:"storage_account_name"`
+}
+
+func (r LocalUserResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.LocalUserName,
+		},
+		"storage_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.StorageAccountName,
+		},
+		"ssh_key_enabled": {
+			Type:         pluginsdk.TypeBool,
+			Optional:     true,
+			Default:      false,
+			AtLeastOneOf: []string{"ssh_key_enabled", "ssh_password_enabled"},
+		},
+		"ssh_password_enabled": {
+			Type:         pluginsdk.TypeBool,
+			Optional:     true,
+			Default:      false,
+			AtLeastOneOf: []string{"ssh_key_enabled", "ssh_password_enabled"},
+		},
+		"home_directory": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"ssh_authorized_key": {
+			Type:         pluginsdk.TypeList,
```

Comment:
API won't return this, so it should be OK to use list.

Reply:
I'm surprised about that since these are Public Keys which should be fine (Private Keys on the other hand I would understand) - is there an API bug for that?

Reply:
Not yet. But is there a specific reason for getting that key from API, given this attribute is a force new one?